model/src/ModelCommandAPI_TEST.cc

```diff
@@ -0,0 +1,330 @@
+ *
+*/
+
+#include "cmd/ModelCommandAPI.hh"
+
+#include <cstdio>
+#include <cstdlib>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include "ignition/gazebo/Server.hh"
+
+static const std::string kIgnModelCommand(std::string(IGN_PATH) +
+                                          "/ign model ");
+
+
+/////////////////////////////////////////////////
+/// \brief Used to avoid the cases where the zero is
+/// represented as a negative number.
+/// \param _text Output string that may have negative zero values.
+void ReplaceNegativeZeroValues(std::string &_text)
+{
+  std::string neg_zero{"-0.000000"};
+  std::string zero{"0.000000"};
+  size_t pos = 0;
+  while ((pos = _text.find(neg_zero, pos)) != std::string::npos)
+  {
+        _text.replace(pos, neg_zero.length(), zero);
+        pos += zero.length();
+  }
+}
+
+/////////////////////////////////////////////////
+std::string customExecStr(std::string _cmd)
+{
+  _cmd += " 2>&1";
+  FILE *pipe = popen(_cmd.c_str(), "r");
+
+  if (!pipe)
+    return "ERROR";
+
+  char buffer[128];
+  std::string result = "";
+
+  while (!feof(pipe))
+  {
+    if (fgets(buffer, 128, pipe) != nullptr)
+    {
+      result += buffer;
+    }
+  }
+
+  pclose(pipe);
+  return result;
+}
+
+// Test `ign model` command when no Gazebo server is running.
+TEST(ModelCommandAPI, NoServerRunning)
+{
+  const std::string cmd = kIgnModelCommand + "--list ";
+  const std::string output = customExecStr(cmd);
+  const std::string expectedOutput =
+        "\nService call to [/gazebo/worlds] timed out\n"
+        "Command failed when trying to get the world name "
+        "of the running simulation.\n";
+  EXPECT_EQ(expectedOutput, output);
+}
+
+// Tests `ign model` command.
+TEST(ModelCommandAPI, Commands)
+{
+  ignition::gazebo::ServerConfig serverConfig;
+  // Using an static model to avoid any movements in the simulation.
+  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
+                          "/test/worlds/static_diff_drive_vehicle.sdf");
+
+  ignition::gazebo::Server server(serverConfig);
+  // Run at least one iteration before continuing to guarantee correctly set up.
+  ASSERT_TRUE(server.Run(true, 5, false));
+  // Run without blocking.
+  server.Run(false, 0, false);
+
+  // Tested command: ign model --list
+  {
+    const std::string cmd = kIgnModelCommand + "--list";
+    const std::string output = customExecStr(cmd);
+    const std::string expectedOutput =
+      "\nRequesting state for world [diff_drive]..."
+      "\n\nAvailable models:\n"
+      "    - ground_plane\n"
+      "    - vehicle_blue\n";
+    EXPECT_EQ(expectedOutput, output);
+  }
+
+  // Tested command: ign model -m vehicle_blue
+  {
+    const std::string cmd = kIgnModelCommand + "-m vehicle_blue";
+    std::string output = customExecStr(cmd);
+    ReplaceNegativeZeroValues(output);
+    const std::string expectedOutput =
+      "\nRequesting state for world [diff_drive]...\n\n"
+      "Name: vehicle_blue\n"
```

Comment:
How about printing the entity for the model? I see the entities for its children below, but not for the model itself.

Reply:
Addressed